shared/pubsub.js

```diff
@@ -0,0 +1,386 @@
+'use strict'
+
+import type { JSONObject, JSONType } from '~/shared/types.js'
+
+// ====== Types ====== //
+
+export type Message = {
+  [key: string]: JSONType,
+  +type: string
```

Comment:
What is this + prefix thing? I've never seen it before. Can you add a comment explaining?

Reply:
It's just the Flow way of telling that a property should be considered read-only after its initialization, so that Flow will report an error if it notice that some code is overwriting the property value.

Reply:
Ah I see. Well, that's pretty advanced flow usage for most people I feel. Could you add a line comment after this, something to the effect of:

```js
  +type: string // the '+' tells flow that this property is read-only after initialization
```

This will help people who are unfamiliar with flow quickly grasp what's going on without having to dive through the depths of the Flowtype documentation.